### PR TITLE
Added the hashing for MD5

### PR DESCRIPTION
**Pull request by Cerfoglg**
Added MD5 hashing for golang, java and python

### CODE REVIEW

**Comment by VincenzoFerme** [comment]
@Cerfoglg needs to be updated to provide projects that can be built and by abstracting the name of the method to hide we internally use MD5.

**Comment by Cerfoglg** [comment]
@VincenzoFerme Fixed this

File: go/golang-hashing/src/hashing/hashing.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+func hashKey(key string) string {
+	return MD5(key)
+	}
+
+func MD5(key string) string {
+	hasher := md5.New()
+    hasher.Write([]byte(key))
+    hashString := hex.EncodeToString(hasher.Sum(nil))
+	return (hashString[:4]+"/"+key)
```
Comment (VincenzoFerme) [inline go/golang-hashing/src/hashing/hashing.go]: Move the number of characters we keep from the MD5 hash, in a variable defining it.

File: java/src/Hashing.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	public static String hashKey(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
+		return Hashing.MD5(key);
+	}
+	
+	public static String MD5(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
+		byte[] bytesOfMessage = key.getBytes("UTF-8");
+		MessageDigest md = MessageDigest.getInstance("MD5");
+		byte[] hashBytes = md.digest(bytesOfMessage);
+		String hashString = new BigInteger(1,hashBytes).toString(16);
+		hashString = hashString.substring(0, 4);
```
Comment (VincenzoFerme) [inline java/src/Hashing.java]: Move the number of characters we keep from the MD5 hash, in a variable defining it.

File: python/hashing.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import hashlib
+
+def hashKey(key):
+    return MD5(key)
+
+def MD5(key):
+    return hashlib.md5(key.encode('utf-8')).hexdigest()[:4]+"/"+key
```
Comment (VincenzoFerme) [inline python/hashing.py]: Move the number of characters we keep from the MD5 hash, in a variable defining it.

**Comment by Cerfoglg** [comment]
@VincenzoFerme  Made it so we have numOfCharacters variable to choose number of hash characters.

**Pull request merged**